mobopt/_target_space.py

```diff
@@ -354,4 +355,66 @@ def lhs_points(self, n_pts):
         for i in range(self.NParam):
             X[:, i] = X[:, i] * (self.pbounds[i][1] - self.pbounds[i][0]) + self.pbounds[i][0]
 
-        return X
\ No newline at end of file
+        return X
+
+    def plot_gp(self, gpr_model, n_eval_pts, title, n_samples=2):
+        """
+        Plot n_samples inside pbounds drawn from the Gaussian process model in input.
+
+        If the Gaussian process model is not trained then the drawn samples are
+        drawn from the prior distribution. Otherwise, the samples are drawn from
+        the posterior distribution.
+        """
+        X = np.asarray(self.lhs_points(n_eval_pts))
+
+        if title.endswith("prior"):
+            y_mean = np.zeros((self.NObj, n_eval_pts))
+            y_std = np.zeros((self.NObj, n_eval_pts))
+            y_samples = np.zeros((self.NObj, n_eval_pts, n_samples))
+
+        elif title.endswith("posterior"):
+            l = self._X.shape[0]
+            y_mean = np.zeros((self.NObj, l + n_eval_pts))
+            y_std = np.zeros((self.NObj, l + n_eval_pts))
+            y_samples = np.zeros((self.NObj, l + n_eval_pts, n_samples))
+            X = np.append(X, self._X, axis=0)
+
+        else:
+            raise ValueError("title must end with 'prior' or 'posterior'")
+
+        for i in range(self.NObj):
+            y_mean[i], y_std[i] = gpr_model[i].predict(X, return_std=True)
+            y_samples[i] = gpr_model[i].sample_y(X, n_samples=n_samples)
+
+        fig, ax = plt.subplots(1, 2, figsize=(20, 10))
+
+        for dim in range(self.NObj):
+
+            for idx in range(n_samples):
+                ax[dim].plot(
+                    np.sort(X[:,0]),
+                    -y_samples[dim, np.argsort(X[:, 0]), idx],
```

Comment:
Potreste salvarvi gli output di `sort` e `argsort` in variabili così non dovete chiamarle ogni volta e rendete il codice più leggibile